src/ui/public/stringify/__tests__/_duration.js

```diff
@@ -0,0 +1,50 @@
+import expect from 'expect.js';
+import ngMock from 'ngMock';
+import RegistryFieldFormatsProvider from 'ui/registry/field_formats';
+
+describe('Duration Format', function () {
+  let fieldFormats;
+  let DurationFormat;
+
+  beforeEach(ngMock.module('kibana'));
+  beforeEach(ngMock.inject(function (Private) {
+    fieldFormats = Private(RegistryFieldFormatsProvider);
+    DurationFormat = fieldFormats.getType('duration');
+  }));
+
+  test({ inputFormat: 'seconds', outputFormat: 'humanize' })
+    (-60, 'minus a minute')
+    (60,  'a minute')
+    (125, '2 minutes');
+
+  test({ inputFormat: 'minutes', outputFormat: 'humanize' })
+    (-60, 'minus an hour')
+    (60,  'an hour')
+    (125, '2 hours');
+
+  test({ inputFormat: 'minutes', outputFormat: 'asHours' })
+    (-60, '-1')
+    (60,  '1')
+    (125, '2');
+
+  test({ inputFormat: 'seconds', outputFormat: 'asSeconds', outputPrecision: 0 })
+    (-60, '-60')
+    (60,  '60')
+    (125, '125');
+
+  test({ inputFormat: 'seconds', outputFormat: 'asSeconds', outputPrecision: 2 })
+    (-60, '-60.00')
+    (-32.333, '-32.33')
+    (60,  '60.00')
+    (125, '125.00');
+
+  function test({ inputFormat, outputFormat, outputPrecision = 0 }) {
```

Comment:
By providing a default value for this, you're actually making it so `DurationFormat` is never tested with `outputPrecision` being `undefined` even though that's a perfectly valid scenario (I think).